doc/src/modules/vector/basics.rst

```diff
@@ -0,0 +1,254 @@
+be initialized in :mod:`sympy.vector` as
+
+  >>> from sympy.vector import CoordSysCartesian
+  >>> N = CoordSysCartesian('N')
+
+The string parameter to the constructor denotes the name assigned to the
+system, and will primarily be used for printing purposes.
+
+Once a coordinate system (in essence, a ``CoordSysCartesian`` instance)
+has been defined, we can access the basis/base vectors (i.e. the 
+:math:`\mathbf{\hat{i}}`, :math:`\mathbf{\hat{j}}` and 
+:math:`\mathbf{\hat{k}}` vectors) and coordinate variables/base 
+scalars (i.e. the :math:`\mathbf{x}`, :math:`\mathbf{y}` and 
+:math:`\mathbf{z}` variables) corresponding to it. We will talk
+about coordinate variables in the later sections.
+
+The basis vectors for the :math:`X`, :math:`Y` and :math:`Z` 
+directions can be accessed using the ``i``, ``j`` and ``k``
+properties respectively.
+
+  >>> N.i
+  N.i
+  >>> type(N.i)
+  <class 'sympy.vector.vector.BaseVector'>
+
+As seen above, the basis vectors are all instances of a class called 
+``BaseVector``.
+
+When a ``BaseVector`` is multiplied by a scalar (essentially any
+SymPy ``Expr``), we get a ``VectorMul`` - the product of
+a base vector and a scalar.
+
+  >>> 3*N.i
+  3*N.i
+  >>> type(3*N.i)
+  <class 'sympy.vector.vector.VectorMul'>
+
+Addition of ``VectorMul`` and ``BaseVectors`` gives rise to
+formation of ``VectorAdd`` - except for special cases, ofcourse.
+
+  >>> v = 2*N.i + N.j
+  >>> type(v)
+  <class 'sympy.vector.vector.VectorAdd'>
+  >>> v - N.j
+  2*N.i
+  >>> type(v - N.j)
+  <class 'sympy.vector.vector.VectorMul'>
+
+What about a zero vector? It can be accessed using the ``zero``
+attribute assigned to class ``Vector``. Since the notion of a zero
+vector remains the same regardless of the coordinate system in 
+consideration, we use ``Vector.zero`` wherever such a quantity is
+required.
+
+  >>> from sympy.vector import Vector
+  >>> Vector.zero
+  0
+  >>> type(Vector.zero)
+  <class 'sympy.vector.vector.VectorZero'>
+  >>> N.i + Vector.zero
+  N.i
+  >>> Vector.zero == 2*Vector.zero
+  True
+
+All the classes shown above - ``BaseVector``, ``VectorMul``, 
+``VectorAdd`` and ``VectorZero`` are subclasses of ``Vector``.
+
+You should never have to instantiate objects of any of the
+subclasses of ``Vector``. Using the ``BaseVector`` instances assigned to a
+``CoordSysCartesian`` instance and (if needed) ``Vector.zero``
+as building blocks, any sort of vectorial expression can be constructed
+with the basic mathematical operators ``+``, ``-``, ``*``.
+and ``/``.
+
+  >>> v = N.i - 2*N.j
+  >>> v/3
+  1/3*N.i + (-2/3)*N.j
+  >>> v + N.k
+  N.i + (-2)*N.j + N.k
+  >>> Vector.zero/2
+  0
+  >>> (v/3)*4
+  4/3*N.i + (-8/3)*N.j
+
+
+In addition to the elementary mathematical operations, the vector 
+operations of ``dot`` and ``cross`` can also be performed on 
+``Vector``.
+
+  >>> v1 = 2*N.i + 3*N.j - N.k
+  >>> v2 = N.i - 4*N.j + N.k
+  >>> v1.dot(v2)
+  -11
+  >>> v1.cross(v2)
+  (-1)*N.i + (-3)*N.j + (-11)*N.k
```

Comment:
I kept it this way so that the distinction between the coefficient and the basis vector parts is clear from looking at the expression itself. The bracketed expressions are always the measure numbers.